drivers/scheduler/dcos/marathon_ops.go

```diff
@@ -0,0 +1,110 @@
+package dcos
+
+import (
+	"regexp"
+	"sync"
+	"time"
+
+	marathon "github.com/gambol99/go-marathon"
+	"github.com/portworx/sched-ops/task"
+)
+
+// Ops is an interface to perform Marathon related operations
+type Ops interface {
+	ApplicationOps
+}
+
+// ApplicationOps is an interface to perform application operations
+type ApplicationOps interface {
+	// CreateApplication creates an application in Marathon
+	CreateApplication(*marathon.Application) (*marathon.Application, error)
+	// ValidateAppliation checks if the aplication is running and healthy
+	ValidateApplication(string) error
+	// DeleteApplication deletes the given application
+	DeleteApplication(string) error
+	// ValidateTerminatedApplication validates if the application is terminated
+	ValidateTerminatedApplication(string) error
+}
+
+var (
+	instance Ops
+	once     sync.Once
+)
+
+type marathonOps struct {
+	client marathon.Marathon
+}
+
+// MarathonClient returns a singleton instance of Marathon Ops type
+func MarathonClient() Ops {
+	once.Do(func() {
+		instance = &marathonOps{}
+	})
+	return instance
+}
+
+// Initialize Marathon client if not initialized
+func (m *marathonOps) initMarathonClient() error {
+	if m.client == nil {
+		marathonURL := "http://192.168.65.90:8080"
+		config := marathon.NewDefaultConfig()
+		config.URL = marathonURL
+
+		client, err := marathon.NewClient(config)
+		if err != nil {
+			return err
+		}
+
+		m.client = client
+	}
+	return nil
+}
+
+func (m *marathonOps) CreateApplication(app *marathon.Application) (*marathon.Application, error) {
+	if err := m.initMarathonClient(); err != nil {
+		return nil, err
+	}
+
+	return m.client.CreateApplication(app)
+}
+
+func (m *marathonOps) ValidateApplication(name string) error {
+	if err := m.initMarathonClient(); err != nil {
+		return err
+	}
+
+	return m.client.WaitOnApplication(name, 5*time.Minute)
```

Comment:
Is this a blocking call?
The callers are not expecting ValidateApplication to be a blocking call. Also the time period can be a constant.

Reply:
Renamed the methods